packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -342,46 +326,50 @@ class _CupertinoAppState extends State<CupertinoApp> {
           data: effectiveThemeData,
           child: Builder(
             builder: (BuildContext context) {
-              return WidgetsApp(
-                key: GlobalObjectKey(this),
-                navigatorKey: widget.navigatorKey,
-                navigatorObservers: _navigatorObservers,
-                pageRouteBuilder: <T>(RouteSettings settings, WidgetBuilder builder) =>
-                  CupertinoPageRoute<T>(settings: settings, builder: builder),
-                home: widget.home,
-                routes: widget.routes,
-                initialRoute: widget.initialRoute,
-                onGenerateRoute: widget.onGenerateRoute,
-                onGenerateInitialRoutes: widget.onGenerateInitialRoutes,
-                onUnknownRoute: widget.onUnknownRoute,
-                builder: widget.builder,
-                title: widget.title,
-                onGenerateTitle: widget.onGenerateTitle,
-                textStyle: CupertinoTheme.of(context).textTheme.textStyle,
-                color: CupertinoDynamicColor.resolve(widget.color ?? effectiveThemeData.primaryColor, context),
-                locale: widget.locale,
-                localizationsDelegates: _localizationsDelegates,
-                localeResolutionCallback: widget.localeResolutionCallback,
-                localeListResolutionCallback: widget.localeListResolutionCallback,
-                supportedLocales: widget.supportedLocales,
-                showPerformanceOverlay: widget.showPerformanceOverlay,
-                checkerboardRasterCacheImages: widget.checkerboardRasterCacheImages,
-                checkerboardOffscreenLayers: widget.checkerboardOffscreenLayers,
-                showSemanticsDebugger: widget.showSemanticsDebugger,
-                debugShowCheckedModeBanner: widget.debugShowCheckedModeBanner,
-                inspectorSelectButtonBuilder: (BuildContext context, VoidCallback onPressed) {
-                  return CupertinoButton.filled(
-                    child: const Icon(
-                      CupertinoIcons.search,
-                      size: 28.0,
-                      color: CupertinoColors.white,
-                    ),
-                    padding: EdgeInsets.zero,
-                    onPressed: onPressed,
-                  );
-                },
-                shortcuts: widget.shortcuts,
-                actions: widget.actions,
+              return InheritedHeroController(
```

Comment:
I am not in love with the name, I don't think we usually put "inherited" in the name of "InheritedWidget" subclasses. But I also don't have a great suggestion for it...

Reply:
Maybe `HeroControllerScope`?